addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -17,6 +17,18 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
         amountToSet: 0,
         giftCardBarcode: "",
       });
+      this.useAutoFocus(this.state);
+    }
+
+    useAutoFocus(state) {
+      function autofocus() {
+        if (state.showBarcodeGeneration) {
+            const elem = document.querySelector(`.giftCardPopupInput`);
```

Comment:
```suggestion
      const component = owl.useComponent();
      function autofocus() {
        if (state.showBarcodeGeneration) {
            const elem = component.el.querySelector(`.giftCardPopupInput`);
```
I think it's better if we only query the input from the component's element. Does it work?

Reply:
Yes it does, I made the change :)

Reply:
And no need to use `onMounted` because `showBarcodeGeneration` starts with `false`, and the time it is set to `true`, `onPatched` callback will be called.

One final comment, about the commit message. Remove `pos` in the commit title because you only modified the `pos_gift_card` addon.

Reply:
You need to modify the commit message, not the title of this PR.

Reply:
My bad, it's changed :)